compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -546,13 +546,27 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             _ => None,
         });
         if let Some(intrinsic) = panic_intrinsic {
+            use rustc_const_eval::might_permit_raw_init::might_permit_raw_init;
             use AssertIntrinsic::*;
+
             let ty = instance.unwrap().substs.type_at(0);
             let layout = bx.layout_of(ty);
             let do_panic = match intrinsic {
                 Inhabited => layout.abi.is_uninhabited(),
-                ZeroValid => !layout.might_permit_raw_init(bx, InitKind::Zero, strict_validity),
-                UninitValid => !layout.might_permit_raw_init(bx, InitKind::Uninit, strict_validity),
+                ZeroValid => !might_permit_raw_init(
+                    bx.tcx(),
+                    source_info.span,
+                    layout,
+                    strict_validity,
```

Comment:
considering this argument is from the single call to `codegen_panic_intrinsic` and is `self.cx.tcx().sess.opts.debugging_opts.strict_init_checks`, I think you could remove the argument everywhere and check the flag directly inside `might_permit_raw_init`